packages/vc-verification/src/issuer-verification-vc.ts

```diff
@@ -143,46 +135,65 @@ export class VCIssuerVerification {
    * @returns
    */
   private async isRoleIssuerDID(role: string) {
-    const issuers = await this._issuerDefResolver.getIssuerDefinition(role);
+    const issuers = await this.issuerResolver.getIssuerDefinition(role);
     return issuers && issuers.issuerType === 'DID';
   }
 
   /**
    * Verifies issuer's authority to issue credential for a namespace
-   * @param {string} namespace
-   * @param {string} issuerDID
-   * @returns boolean
+   * @param {string} role role credential name
+   * @param {string} issuer DID of issuer
    */
-  async verifyIssuerAuthority(
-    namespace: string,
-    issuerDID: string
-  ): Promise<boolean> {
-    const issuers = await this._issuerDefResolver.getIssuerDefinition(
-      namespace
-    );
-    if (issuers && issuers.did && issuers.did.length > 0) {
-      for (let i = 0; i < issuers.did.length; i++) {
-        const issuerAddr = addressOf(issuerDID);
-        const roleIssuerAddr = addressOf(issuers.did[i]);
-        if (roleIssuerAddr.toUpperCase() === issuerAddr.toUpperCase()) {
-          return true;
-        }
-      }
+  async verifyIssuer(issuer: string, role: string) {
+    const issuers = await this.issuerResolver.getIssuerDefinition(role);
+    if (!issuers) {
+      throw new NoIssuers(role);
     }
-    let vc;
-    if (issuers && issuers.roleName) {
-      vc = await this._credentialResolver.getCredential(
-        issuerDID,
-        issuers.roleName
-      );
+    if (issuers.issuerType === 'DID' && issuers.did) {
+      if (!issuers.did?.some((i) => i.toUpperCase() === issuer.toUpperCase())) {
+        throw new IssuerNotAuthorized(
+          issuer,
+          role,
+          'issuer is not in DID list'
+        );
+      }
+    } else if (issuers.issuerType === 'ROLE' && issuers.roleName) {
+      try {
+        await this.verifyCredential(issuer, issuers.roleName);
+      } catch (e) {
+        throw new IssuerNotAuthorized(issuer, role, (<Error>e).message);
+      }
+    } else {
+      throw new InvalidIssuerType(role, issuers.issuerType);
     }
-    if (!vc) {
-      return false;
+  }
+
+  /**
+   * Verifies that role credential was issued to subject by authorized issuer
+   * @param subject DID of revoker
+   * @param role fully qualified role credential name
+   *
+   * @returns verified role credential
+   */
+  async verifyCredential(
```

Comment:
@JGiter I like what you've done here with recursive approach of `verifyCredential` calling `verifyIssuer`, however, I think that it has introduces some duplication with `verifyChainOfTrust` as `verifyChainOfTrust` and `verifyIssuer` essentially do the same thing of "Verifying that role credential was issued to subject by authorized issuer".

IMO, the `verifyCredential`/`verifyIssuer` recursion approach is easier to read than the `verifyChainOfTrust` code, so we should remove the `verifyChainOfTrust`/`verifyChainOfTrustByRoleDefinition`/`verifyChainOfTrustCallback` code.

How does this sound @JGiter @nichonien ?

Reply:
It does seem like a good approach but there are some flaws:

1. I would want to have two separate function `verifyCredential` and `verifyIssuer` for verifying the credential and issuer respectively. Since, `ICL` and `SSI-Hub` may need such functionality for individual calls.
2. By definition `verifyCredential` should just verify the credential and probably it could take a `callbackfunction` if needed for flexibility / choice of verification.
3. I would keep `verifyChainOfTrustByRoleDefinition ` and `verifyChainOfTrustCallback ` to verify entire hierarchy of issuance based on the presented credential.
4. Initially the `verifyChainOfTrust` was supposed to be the entry point for chain-of-trust verification based on the passed parameters, but I guess it is not needed now, we can directly call the specific functions or maybe mark them private and export only `verifyChainOfTrust`. 
5. I agree the naming convention could be improved and some code refactoring as well.

Reply:
Yes, we can remove verifyChainOfTrust. It is not used anywhere and its logic is implemented in `verifiy..` methods

Reply:
We can also do the same in issuer-verification-offchain